doc/developer/design/20240502_deprecate_subsources.md

```diff
@@ -0,0 +1,190 @@
+# sql+storage: deprecate concept of subsources
+
+- Associated:
+    - [storage/sources: make (sub)sources a flavor of tables #20208
+      ](https://github.com/MaterializeInc/materialize/issues/20208)
+    - [sql/storage: consider if TEXT COLUMNS, IGNORE COLUMNS belongs on
+      subsources
+      #26774](https://github.com/MaterializeInc/materialize/issues/26774)
+
+## The Problem
+
+There are two distinct notions of `SUBSOURCE`s in MZ, both of which can be
+folded neatly into other concepts.
+
+- Ingestion export subsources should be tables (as should the current primary
+  source's collection)
+- Progress subsources should become the relation returned when querying sources
+
+Doing both of these lets us remove the concept of subsources from Materialize
+entirely. It, by definition, also lets us get rid of the notion of "progress
+subsources," which is a concept that users often find confusion.
+
+## Success Criteria
+
+- Users no longer need to use the keyword `SUBSOURCE` to interact with
+  Materialize in any capacity; and we do this without deprecating any
+  functionality whatsoever.
```

Comment:
I think this is implied, but might be worth explicitly stating that we're also going to remove `subsource` from any system catalog table names or rows. (Since these aren't quite subsource _keywords_.)